java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageReader.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.ipc.message;
+
+
+import io.netty.buffer.ArrowBuf;
+import org.apache.arrow.flatbuf.Message;
+import org.apache.arrow.memory.BufferAllocator;
+
+import java.io.IOException;
+
+public interface MessageReader {
+
+  Message readNextMessage() throws IOException;
+
+  ArrowBuf readMessageBody(Message message, BufferAllocator allocator) throws IOException;
```

Comment:
Maybe add a bit doc of what these methods are supposed to do? It's not very clear how to use `readNextMessage` and `readMessageBody`

Reply:
Yeah, I meant to say that I still need to go through these changes and make sure everything is documented properly.